CMakeLists.txt

```diff
@@ -13,31 +13,53 @@ project(FORMAT_BENCHMARKS)
 set(CMAKE_MACOSX_RPATH ON)
 set(CMAKE_CXX_STANDARD 17)
 
+include(CheckCXXCompilerFlag)
+
 # Workaround a JCC bug in Intel CPUs:
 # https://www.intel.com/content/dam/support/us/en/documents/processors/
 # mitigations-jump-conditional-code-erratum.pdf
 # to get more reliable benchmark results.
-set(INTEL FALSE)
-if (APPLE)
-  execute_process(COMMAND sysctl -n machdep.cpu.brand_string
-                  OUTPUT_VARIABLE out)
-  if (out MATCHES "Intel.*")
-    set(INTEL TRUE)
-  endif ()
-else ()
-  file(READ /proc/cpuinfo out)
-  if (out MATCHES "(.|\n)*GenuineIntel(.|\n)*")
-    set(INTEL TRUE)
+
+if (NOT DEFINED INTEL)
+  if (APPLE)
+    execute_process(COMMAND sysctl -n machdep.cpu.brand_string
+                    OUTPUT_VARIABLE out)
+    if (out MATCHES "Intel.*")
+      option(INTEL "Are we compiling on/for an intel target?" ON)
```

Comment:
I suggest defining the option in one place to reduce copy-paste.

Reply:
Thing is, I would then need to call `set( ... CACHE ... FORCE)` which requires the comment too?
I can however put the comment in a variable to avoid the string duplication

Reply:
Scratch that, I'll just set an intermediary variable with the value and the `option` call at the end. Sorry I'm just tired!